connector.go

```diff
@@ -0,0 +1,264 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/UniversityRadioYork/baps3-go"
+)
+
+// GetResponse is the outer structure of all GET responses.
+type GetResponse struct {
+	Status string
+	Value  interface{}
+}
+
+// GetOk creates a GetResponse wrapping a successful GET result.
+func GetOk(value interface{}) *GetResponse {
+	r := new(GetResponse)
+	r.Status = "ok"
+	r.Value = value
+	return r
+}
+
+type bfConnector struct {
+	conn   *baps3.Connector
+	name   string
+	wg     *sync.WaitGroup
+	logger *log.Logger
+	state  *serviceState
+
+	reqCh chan httpRequest
+	resCh <-chan baps3.Message
+
+	// TODO(CaptainHayashi): move this away from baps3.Message to
+	// something generic.
+	updateCh chan<- baps3.Message
+}
+
+func initBfConnector(name string, updateCh chan baps3.Message, waitGroup *sync.WaitGroup, logger *log.Logger) (c *bfConnector) {
```

Comment:
could updateCh get a direction to it? it's only being used in one direction (from the bfConnector), right?